src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
+	roomId: string,
 ) {
-	socket.emit(emitterName, message, roomId, userName)
+	socket.emit(emitterName, message, userName, timeStamp, roomId)
+}
+
+export function useAllUsersPointsListener(listenerName: string) {
```

Comment:
question: Why did you create this function which is exactly the same as `useSocketListener()`? The only difference is the type passed to `listenerRes`. In one is `AllUsersPointsRes` and in another is `ListenerRes`. Are you fighting TS here?

Reply:
100% TypeScript is fighting ME. I can't fix it, Copilot can't fix it, googling doesn't provide an answer. So I created 2 versions so it would work. If you can help solve this, I'm 100% open to having only one.

Reply:
Here's [a loom](https://www.loom.com/share/55e4419c46f14e79ad3a081ecbce4b57?sid=d59c6d74-3228-49ff-8d5a-45dfbedf7730) explaining it. (the password is sent on our chat).

TLDR: To override types you use `as unknown as YourType`. This tells TS "_Look, trust me here. I know the original type is different, but I want you to pretend you don't know and this is the new type._"

```js
useSocketListener('all-users-story-points', (allUserPoints) => {
  const allUserPointsMessages = (allUserPoints as unknown as AllListenerRes).message
  // ...
}
```

Reply:
Video watched. I'm making those changes, eliminating the duplicate listener and the string "booleans". Thanks for that video!!

Reply:
DONE